service/frontend/workflow_handler.go

```diff
@@ -2092,69 +2092,90 @@ func (wh *WorkflowHandler) ListOpenWorkflowExecutions(ctx context.Context, reque
 		return nil, errRequestNotSet
 	}
 
+	maxPageSize := int32(wh.config.VisibilityMaxPageSize(request.GetNamespace()))
+	if request.GetMaximumPageSize() <= 0 || request.GetMaximumPageSize() > maxPageSize {
+		request.MaximumPageSize = maxPageSize
+	}
+
+	namespaceName := namespace.Name(request.GetNamespace())
+	namespaceID, err := wh.namespaceRegistry.GetNamespaceID(namespaceName)
+	if err != nil {
+		return nil, err
+	}
+
+	baseReq := &manager.ListWorkflowExecutionsRequestV2{
+		NamespaceID:   namespaceID,
+		Namespace:     namespaceName,
+		PageSize:      int(request.GetMaximumPageSize()),
+		NextPageToken: request.NextPageToken,
+		Query: fmt.Sprintf(
+			"%s = '%s'",
+			searchattribute.ExecutionStatus,
+			enumspb.WORKFLOW_EXECUTION_STATUS_RUNNING,
+		),
+	}
+
 	if request.StartTimeFilter == nil {
 		request.StartTimeFilter = &filterpb.StartTimeFilter{}
 	}
 
 	if request.StartTimeFilter.EarliestTime == nil || request.StartTimeFilter.EarliestTime.AsTime().IsZero() {
 		request.StartTimeFilter.EarliestTime = timestamppb.New(minTime)
+	} else {
+		baseReq.Query += fmt.Sprintf(
+			"AND %s >= '%s'",
```

Comment:
Missing space in the front. Wondering if it would be better to create a slice of filters and then do a join.

```
filters := []string{}
filters = append(filters, fmt.Sprintf("%s = '%s'", ...))
...
strings.Join(filters, " AND ")
```

```suggestion
			" AND %s >= '%s'",
```